src/network/message.rs

```diff
@@ -358,10 +355,15 @@ impl Decodable for HeaderDeserializationWrapper {
         }
         Ok(HeaderDeserializationWrapper(ret))
     }
+
+    #[inline]
+    fn consensus_decode<D: io::Read>(d: D) -> Result<Self, encode::Error> {
+        Self::consensus_decode_from_finite_reader(d.take(MAX_MSG_SIZE as u64))
```

Comment:
Oh, now I see that the previous code was using `MAX_VEC_SIZE`. I was wondering if messages on the wire bound by the same limit as block.